lib/parse.js

```diff
@@ -919,13 +925,21 @@ function parse($TEXT, options) {
         if (!is("punc", ";")) {
             init = is("keyword", "var")
                 ? (next(), var_(true))
-                : expression(true, true);
+                : is("keyword", "let")
```

Comment:
And in strict mode, those 3 are all failed because `let` is an strict mode reserved word.

The draft of ES2015 has so much changes, I think some of code refactoring need to be done in order to support them. 
but for people who is carefully using the generator, let and for-of, I think this patch might be work for them.